### ISSUE TITLE
static podman :)

### ISSUE THREAD

**Issue by pwFoo** [mgoltzsche/podman-static#1]
Hi, 
searched for podman static binary and found your repository! Sounds great!

Are that the _minimal_ dependencies?
```
# Install iptables & new-uidmap
RUN apk add --no-cache ca-certificates iptables ip6tables shadow-uidmap
# Copy binaries from other images
COPY --from=runc   /usr/local/bin/runc   /usr/local/bin/runc
COPY --from=podman /usr/local/bin/podman /usr/local/bin/podman
COPY --from=conmon /usr/libexec/podman/conmon /usr/libexec/podman/conmon
COPY --from=cniplugins /usr/libexec/cni /usr/libexec/cni
COPY --from=skopeo /usr/local/bin/skopeo /usr/local/bin/skopeo
COPY --from=fuse-overlayfs /usr/bin/fuse-overlayfs /usr/local/bin/fuse-overlayfs
COPY --from=slirp4netns /slirp4netns/slirp4netns /usr/local/bin/slirp4netns
COPY --from=buildah /usr/local/bin/buildah /usr/local/bin/buildah
```

Or could buildah, skopeo (, ...?) dropped to just use podman (pull and run images)?

**Comment by mgoltzsche** [comment]
@pwFoo You can drop buildah & skopeo if you don't want to build images. Also you can drop all CNI plugin(s) you don't need.
If you intent to run rootless containers you may need fuse-overlayfs & slirp4netns.

**Comment by pwFoo** [comment]
@mgoltzsche
Which cni plugins are needed to run a (docker) container with a normal network / nic and Internet access?

**Comment by mgoltzsche** [comment]
When you run the container **as root** it requires the CNI plugins of the CNI config (`/etc/cni/net.d/`) you provide. The default config within this image is in `/etc/cni/net.d/99-bridge.conflist` and requires the plugins `bridge`, `host-local` & `portmap` (located in `/usr/libexec/cni`).

When you run a **rootless** container podman does not require any CNI plugin since it uses `slirp4netns` directly to emulate a container network on a tap device. Please note that this still has some limitations concerning the communication outside the pod's network namespace (see https://github.com/containers/libpod/issues/1733).

**Comment by pwFoo** [comment]
Built my minimal version by 

- Remove binaries
 ```WORKDIR /mnt/usr/libexec/cni

RUN     rm -f \
        flannel \
        ipvlan \
        macvlan \
        ptp \
        vlan \
        dhcp \
        tuning```
 ```#COPY   --from=skopeo /usr/local/bin/skopeo /mnt/usr/local/bin/skopeo                                                                                
#COPY   --from=fuse-overlayfs /usr/bin/fuse-overlayfs /mnt/usr/local/bin/fuse-overlayfs                                                              
#COPY   --from=slirp4netns /slirp4netns/slirp4netns /mnt/usr/local/bin/slirp4netns                                                                   
#COPY   --from=buildah /usr/local/bin/buildah /mnt/usr/local/bin/buildah```
- add libpod.conf
 ```RUN     wget -O /mnt/etc/containers/libpod.conf https://raw.githubusercontent.com/containers/libpod/36df2e70738ab277699e8240f62355a67a6b2edb/libpod.conf
RUN     sed -i 's/cgroup_manager =.*/cgroup_manager = "cgroupfs"/' /mnt/etc/containers/libpod.conf
RUN     sed -i 's/#namespace = ""/namespace = "default"/' /mnt/etc/containers/libpod.conf```

**Comment by mgoltzsche** [comment]
@pwFoo Unfortunately I didn't found the time to build minimal versions (because then you would need multiple).
However to get a minimal image for your use case you could copy the binaries you need into a new image. Removing files from an existing image will not make it smaller since they are just marked as removed but still take space within the parent layer in which they were added.

**Comment by mgoltzsche** [comment]
fyi: `mgoltzsche/podman:2.1` is now way smaller since I removed some unused stuff:
* Reduced CNI plugins to those required by the default config (loopback, bridge, host-local, portmap) when run as root.
* buildah has been removed since it is now part of the podman binary.

Though slirp4netns and fuse-overlayfs remain within the image since running rootless is a common use case (I wonder if I should rather remove all CNI plugins from the image at some point).
There's now also a version published using `crun` instead of `runc` (`<PODMANVERSION>-crun`).
